packages/block-editor/src/components/rich-text/index.js

```diff
@@ -2,1112 +2,128 @@
+	onSelectionChange,
+	multiline,
+	onTagNameChange,
+	inlineToolbar,
+	wrapperClassName,
+	className,
+	autocompleters,
+	onReplace,
+	onRemove,
+	onMerge,
+	onSplit,
+	isCaretWithinFormattedText,
+	onEnterFormattedText,
+	onExitFormattedText,
+	canUserUseUnfilteredHTML,
+	isSelected: originalIsSelected,
+	onCreateUndoLevel,
+	placeholder,
+	keepPlaceholderOnFocus,
+	// From experimental filter.
+	...experimentalProps
+} ) {
+	let adjustedValue = originalValue;
+	let adjustedOnChange = originalOnChange;
+
+	// Handle deprecated format.
+	if ( Array.isArray( originalValue ) ) {
+		adjustedValue = children.toHTML( originalValue );
+		adjustedOnChange = ( newValue ) => originalOnChange( children.fromDOM(
+			__unstableCreateElement( document, newValue ).childNodes
+		) );
+	}
+
+	return (
+		<RichText
+			{ ...experimentalProps }
+			value={ adjustedValue }
+			onChange={ adjustedOnChange }
+			selectionStart={ selectionStart }
+			selectionEnd={ selectionEnd }
+			onSelectionChange={ onSelectionChange }
+			tagName={ tagName }
+			wrapperClassName={ classnames( wrapperClasses, wrapperClassName ) }
+			className={ classnames( classes, className ) }
+			placeholder={ placeholder }
+			keepPlaceholderOnFocus={ keepPlaceholderOnFocus }
+			__unstableIsSelected={ originalIsSelected }
+			__unstablePatterns={ getPatterns() }
+			__unstableEnterPatterns={ getEnterPatterns() }
+			__unstablePasteHandler={ pasteHandler }
+			__unstableAutocomplete={ Autocomplete }
+			__unstableAutocompleters={ autocompleters }
+			__unstableOnReplace={ onReplace }
+			__unstableOnRemove={ onRemove }
+			__unstableOnMerge={ onMerge }
+			__unstableOnSplit={ onSplit }
+			__unstableMultiline={ multiline }
+			__unstableIsCaretWithinFormattedText={ isCaretWithinFormattedText }
+			__unstableOnEnterFormattedText={ onEnterFormattedText }
+			__unstableOnExitFormattedText={ onExitFormattedText }
+			__unstableCanUserUseUnfilteredHTML={ canUserUseUnfilteredHTML }
+			__unstableOnCreateUndoLevel={ onCreateUndoLevel }
+		>
+			{ ( { isSelected, value, onChange } ) =>
+				<>
+					{ isSelected && multiline === 'li' && (
+						<ListEdit
+							onTagNameChange={ onTagNameChange }
+							tagName={ tagName }
+							value={ value }
+							onChange={ onChange }
+						/>
 					) }
-				</Autocomplete>
-				{ isSelected && <RemoveBrowserShortcuts /> }
-			</div>
-		);
-	}
+					{ isSelected && ! inlineToolbar && (
+						<BlockFormatControls>
+							<FormatToolbar />
+						</BlockFormatControls>
+					) }
+					{ inlineToolbar && (
+						<IsolatedEventContainer>
+							<FormatToolbar />
+						</IsolatedEventContainer>
+					) }
+					{ isSelected && <RemoveBrowserShortcuts /> }
+				</>
+			}
+		</RichText>
+	);
 }
 
-RichText.defaultProps = {
-	formattingControls: [ 'bold', 'italic', 'link', 'strikethrough' ],
-	format: 'string',
-	value: '',
-};
-
 const RichTextContainer = compose( [
 	withInstanceId,
 	withBlockEditContext( ( { clientId } ) => ( { clientId } ) ),
```

Comment:
Minor: The withSelect here, could become a `useSelect`

Reply:
Yes, I've merely moved this code, I don't want to do a rewrite here to avoid making the PR more complex. I'm hoping for a quick merge so I don't have to do painful rebases. :)